src/backend/xacheck.go

```diff
@@ -195,29 +223,47 @@ func (xc *XaCheck) commitRetryBackends(query string, scatter *Scatter, xid strin
 	}
 
 	if len(needCommitBackends) == 0 {
-		log.Info("xacheck.commitRetryBackends.recover.query[%v].find.no.needCommitBackends", query)
+		log.Info("xacheck.commitRetryBackends.recover.query[%v].find.no.Backends.need.retry.[%v]", query, retry.Times)
 	}
 
 	// the 2nd stage: xa commit/rollback '$xid'
-	ExecuteOKCount := 0
+	count := 0
 	for _, backend := range needCommitBackends {
 		_, err = txn.ExecuteOnThisBackend(backend, query)
 		if err == nil {
 			log.Info("xacheck.commitRetryBackends.query[%v].success.backend[%v]", query, backend)
-			ExecuteOKCount++
+			count++
 		} else {
 			log.Warning("xacheck.commitRetryBackends.query[%v].backend[%v].error[%T]:%+v", query, backend, err, err)
 		}
 	}
 
-	// if there are items need to retry, and ExecuteOKCount is equal to the needCommitBackends, return true
-	if ExecuteOKCount > 0 && ExecuteOKCount == len(needCommitBackends) {
+	// if there are XA need to retry, and the count committed is equal to the needCommitBackends, return true.
+	if count > 0 && count == len(needCommitBackends) {
 		return true, nil
 	}
+
+	if retry.Times <= 0 {
+		data, err := json.Marshal(retry)
```

Comment:
Here we need a WARNING log for the retry times out.

Reply:
The log is added.